File: java-example/src/test/java/ru/stqa/training/selenium/RegistrationPage.java
package ru.stqa.training.selenium;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class RegistrationPage extends Page {

    public RegistrationPage(WebDriver driver) {
        super(driver);
    }

    public void open() {
        driver.get("http://localhost/litecart/en/create_account");
    }

    public WebElement firstnameInput() {
        return driver.findElement(By.name("firstname"));
    }

    public WebElement lastnameInput() {
        return driver.findElement(By.name("lastname"));
    }
    public WebElement address1Input() {
        return driver.findElement(By.name("address1"));
    }
    public WebElement postcodeInput() {
        return driver.findElement(By.name("postcode"));
    }
    public WebElement cityInput() {
        return driver.findElement(By.name("city"));
    }
    public WebElement emailInput() {
        return driver.findElement(By.name("email"));
    }
    public WebElement phoneInput() {
        return driver.findElement(By.name("phone"));
    }
    public WebElement passwordInput() {
        return driver.findElement(By.name("password"));
    }
    public WebElement confirmedPasswordInput() {
        return driver.findElement(By.name("confirmed_password"));
    }

    public WebElement createAccountButton() {
        return driver.findElement(By.name("create_account"));
    }
}

File: csharp-example/csharp-example/pages/RegistrationPage.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Support.PageObjects;

namespace csharp_example
{
    internal class RegistrationPage : Page
    {
        public RegistrationPage(IWebDriver driver) : base(driver)
        {
            PageFactory.InitElements(driver, this);
        }

        internal void Open()
        {
            driver.Url = "http://localhost/litecart/en/create_account";
        }

        [FindsBy(How = How.Name, Using = "address1")]
        internal IWebElement Address1Input;
        [FindsBy(How = How.Name, Using = "city")]
        internal IWebElement CityInput;
        [FindsBy(How = How.Name, Using = "confirmed_password")]
        internal IWebElement ConfirmedPasswordInput;
        [FindsBy(How = How.Name, Using = "email")]
        internal IWebElement EmailInput;
        [FindsBy(How = How.Name, Using = "firstname")]
        internal IWebElement FirstnameInput;
        [FindsBy(How = How.Name, Using = "lastname")]
        internal IWebElement LastnameInput;
        [FindsBy(How = How.Name, Using = "password")]
        internal IWebElement PasswordInput;
        [FindsBy(How = How.Name, Using = "phone")]
        internal IWebElement PhoneInput;
        [FindsBy(How = How.Name, Using = "postcode")]
        internal IWebElement PostcodeInput;

        [FindsBy(How = How.Name, Using = "create_account")]
        internal IWebElement CreateAccountButton;

        internal void SelectCountry(string country)
        {
            driver.FindElement(By.CssSelector("[id ^= select2-country_code]")).Click();
            driver.FindElement(By.CssSelector(
                    String.Format(".select2-results__option[id $= {0}]", country))).Click();
        }

        internal void SelectZone(string zone)
        {
            wait.Until(d => d.FindElement(
                    By.CssSelector(String.Format("select[name=zone_code] option[value={0}]", zone))));
            new SelectElement(driver.FindElement(By.CssSelector("select[name=zone_code]"))).SelectByValue(zone);
        }

    }
}